Source/Task/AsyncLib.cpp

```diff
@@ -759,6 +759,13 @@ STDAPI ScheduleAsync(
         RETURN_HR(E_UNEXPECTED);
     }
 
+    // Hold another ref here to prevent async call from finishing then triggering dtor before 
+    // we had a chance to call addRef after submission.
+    AsyncStateRef holder;
+    {
+        AsyncBlockInternalGuard internal{ asyncBlock };
+        holder = internal.GetState();
+    }
```

Comment:
This will work, but there is a better solution I believe.

You should change the call below to state->AddRef() to state->Detach(), the net effect is the same (we "leak" one reference for the work callback), but it follows a more common pattern for this kind of thing.